httpmw/logging_mw.go

```diff
@@ -0,0 +1,70 @@
+package httpmw
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+type Logger interface {
+	Info(msg string, keysAndValues ...interface{})
+	Error(msg string, keysAndValues ...interface{})
+}
+
+func NewLogging(logger Logger) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return &loggingMiddleware{
+			next:   next,
+			now:    time.Now,
+			logger: logger,
+		}
+	}
+}
+
+var _ http.Handler = &loggingMiddleware{}
+
+type loggingMiddleware struct {
+	next   http.Handler
+	now    func() time.Time
+	logger Logger
+}
+
+type responseWriter struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (rw *responseWriter) WriteHeader(code int) {
```

Comment:
rw typically written as w. rw make me feel it is read write.

Reply:
Oops, yeah that is supposed to be w. Made the change.